pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1381,7 +1391,9 @@ def test_merge_readonly(self):
             np.arange(20).reshape((5, 4)) + 1, columns=["a", "b", "x", "y"]
         )
 
-        data1._mgr.blocks[0].values.flags.writeable = False
+        for arr in data1._mgr.arrays:
```

Comment:
cabn you comment on this (I know what you are doing by seeing what you added, but is non-obvious here i think)